src/view-compiler.js

```diff
@@ -158,7 +158,7 @@ export class ViewCompiler {
       return this._compileElement(node, resources, instructions, parentNode, parentInjectorId, targetLightDOM);
     case 3: //text node
       //use wholeText to retrieve the textContent of all adjacent text nodes.
-      let expression = resources.getBindingLanguage(this.bindingLanguage).inspectTextContent(resources, node.wholeText);
+      let expression = resources.getBindingLanguage(this.bindingLanguage).inspectTextContent(resources, node.wholeText || node.data);
```

Comment:
There's a few options- we could do a one-time check in the templating package. We could expose this in the PAL. We could polyfill the feature in jsdom. We could PR the fix for jsdom.

Definitely would be more comfortable with your `node.wholeText !== undefined ? node.wholeText : node.data` approach.